pkg/cache/v3/snapshot.go

```diff
@@ -199,7 +109,7 @@ func (s *Snapshot) GetResources(typeURL string) map[string]types.Resource {
 }
 
 // GetResourcesAndTtl selects snapshot resources by type, returning the map of resources and the associated TTL.
-func (s *Snapshot) GetResourcesAndTtl(typeURL string) map[string]types.ResourceWithTTL { // nolint:golint,revive
+func (s *Snapshot) GetResourcesAndTTL(typeURL string) map[string]types.ResourceWithTTL { // nolint:golint,revive
```

Comment:
Nit: remove the `nolint`?

Reply:
Should this, and the other APIs that take a typeURL argument use `resource.Type` for it?

Reply:
I just tried that and it seemed to also be too intrusive in the cache. It would require a lot of type casting again

Reply:
Actually nevermind, just figured something out I will push a big change haha